components/operator-service/src/main/java/com/wso2telco/dep/operatorservice/dao/BlackListWhiteListDAO.java

```diff
@@ -178,35 +196,210 @@ public void blacklist(MSISDNValidationDTO msisdns, final String apiID, final Str
 		return returnList;
 	}
 
-	public String[] getBlacklisted(MSISDNSearchDTO searchDTO) throws Exception {
+	/**
+	 * Retrieves the number of blacklist entries for a particular API ID
+	 * @return
+	 * @throws Exception
+	 */
+	public int getBlacklistCountByApi(BlacklistWhitelistDTO dto) throws Exception {
+		StringBuilder sql = new StringBuilder();
 
+		sql.append("SELECT count(*) as count");
+		sql.append(" FROM ");
+		sql.append(OparatorTable.API_BLACKLIST_WHITELIST.getTObject());
+		sql.append(" WHERE 1=1 ");
+		sql.append(" AND API_ID LIKE ? ");
+		sql.append(" AND APP_ID LIKE ? ");
+		sql.append(" AND ACTION = ? ");
+		sql.append(" AND SERVICE_PROVIDER LIKE ? ");
+
+		Connection conn = null;
+		PreparedStatement ps = null;
+		ResultSet rs = null;
+		int count=0;
+
+		try {
+			conn = DbUtils.getDbConnection(DataSourceNames.WSO2AM_STATS_DB);
+			ps = conn.prepareStatement(sql.toString());
+			ps.setString(1, dto.getApiID());
+			ps.setString(2,dto.getAppId());
+			ps.setString(3,dto.getAction());
+			ps.setString(4,dto.getServiceProvider());
+			rs = ps.executeQuery();
+
+			while (rs.next()) {
+				count = rs.getInt("count");
+			}
+
+		} catch (SQLException e) {
+			throw e;
+		} finally {
+			DbUtils.closeAllConnections(ps, conn, rs);
+		}
+
+		return count;
+	}
+
+	public void streamBlacklistByApi(BlacklistWhitelistDTO dto,int offset, int limit, OutputStream out) throws Exception {
 		StringBuilder sql = new StringBuilder();
 
-		sql.append("SELECT MSISDN,API_ID,API_NAME,USER_ID");
+		sql.append("SELECT apis.API_NAME, apps.NAME, apps.CREATED_BY, bw.MSISDN, bw.ACTION, bw.user, bw.CREATED, bw.LAST_MODIFIED");
```

Comment:
Too much operation in single execution. as previously mentioned better to move to separate file